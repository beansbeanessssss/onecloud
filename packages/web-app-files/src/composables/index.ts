export * from './actions'
export * from './resourcesViewDefaults'
export * from './shares'
