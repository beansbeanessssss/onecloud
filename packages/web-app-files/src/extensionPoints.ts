import { ExtensionPoint } from '@ownclouders/web-pkg'
import { computed } from 'vue'

export const uploadMenuExtensionPoint: ExtensionPoint = {
  id: 'app.files.upload-menu',
  extensionType: 'action',
  multiple: true
}
export const quickActionsExtensionPoint: ExtensionPoint = {
  id: 'app.files.quick-actions',
  extensionType: 'action',
  multiple: true
}
export const batchActionsExtensionPoint: ExtensionPoint = {
  id: 'app.files.batch-actions',
  extensionType: 'action',
  multiple: true
}
export const contextActionsExtensionPoint: ExtensionPoint = {
  id: 'app.files.context-actions',
  extensionType: 'action',
  multiple: true
}
export const fileSideBarExtensionPoint: ExtensionPoint = {
  id: 'app.files.sidebar',
  extensionType: 'sidebarPanel',
  multiple: true
}
export const folderViewsFolderExtensionPoint: ExtensionPoint = {
  id: 'app.files.folder-views.folder',
  extensionType: 'folderView'
}
export const folderViewsFavoritesExtensionPoint: ExtensionPoint = {
  id: 'app.files.folder-views.favorites',
  extensionType: 'folderView'
}
export const folderViewsProjectSpacesExtensionPoint: ExtensionPoint = {
  id: 'app.files.folder-views.project-spaces',
  extensionType: 'folderView'
}

export const extensionPoints = () => {
  return computed<ExtensionPoint[]>(() => {
    return [
      uploadMenuExtensionPoint,
      quickActionsExtensionPoint,
      batchActionsExtensionPoint,
      contextActionsExtensionPoint,
      fileSideBarExtensionPoint,
      folderViewsFolderExtensionPoint,
      folderViewsFavoritesExtensionPoint,
      folderViewsProjectSpacesExtensionPoint
    ]
  })
}
