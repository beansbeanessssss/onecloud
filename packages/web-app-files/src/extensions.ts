import {
  ApplicationSetupOptions,
  Extension,
  useStore,
  useRouter,
  useSearch,
  useFileActionsShowShares,
  useFileActionsCreateQuickLink
} from '@ownclouders/web-pkg'
import { computed, unref } from 'vue'
import { SDKSearch } from './search'

export const extensions = ({ applicationConfig }: ApplicationSetupOptions) => {
  const store = useStore()
  const router = useRouter()
  const { search: searchFunction } = useSearch()

  const { actions: showSharesActions } = useFileActionsShowShares()
  const { actions: quickLinkActions } = useFileActionsCreateQuickLink()

  return computed(
    () =>
      [
        {
          id: 'com.github.owncloud.web.files.search',
          type: 'search',
          searchProvider: new SDKSearch(store, router, searchFunction)
        },
        {
          id: 'com.github.owncloud.web.files.quick-action.collaborator',
          scopes: ['files', 'files.quick-action'],
          type: 'action',
          action: unref(showSharesActions)[0]
        },
        {
          id: 'com.github.owncloud.web.files.quick-action.quicklink',
          scopes: ['files', 'files.quick-action'],
          type: 'action',
          action: unref(quickLinkActions)[0]
        }
      ] satisfies Extension[]
  )
}
