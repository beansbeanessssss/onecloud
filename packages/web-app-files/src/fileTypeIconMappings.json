{
  "7z": "file-zip",
  "accdb": "file",
  "ai": "image",
  "apk": "file-zip",
  "bash": "file-code",
  "bz2": "file-zip",
  "cb7": "file-text",
  "cba": "file-text",
  "cbr": "file-text",
  "cbt": "file-text",
  "cbtc": "file-text",
  "cbz": "file-text",
  "cc": "file-code",
  "cdr": "image",
  "cpp": "file-code",
  "css": "file-code",
  "csv": "file-excel-2",
  "cvbdl": "file-text",
  "c": "file-code",
  "c++": "file-code",
  "deb": "file-zip",
  "doc": "file-word-2",
  "docm": "file-word-2",
  "docx": "file-word-2",
  "dot": "file-word-2",
  "dotx": "file-word-2",
  "eml": "file-text",
  "eot": "image",
  "epub": "file-text",
  "eps": "image",
  "gz": "file-zip",
  "gzip": "file-zip",
  "h": "file-code",
  "hh": "file-code",
  "hpp": "file-code",
  "html": "file-code",
  "htm": "file-code",
  "java": "file-code",
  "js": "file-code",
  "json": "file-code",
  "jpeg": "image",
  "jpg": "image",
  "lwp": "file-word-2",
  "mdb": "file",
  "odp": "file-ppt-2",
  "ods": "file-excel-2",
  "odt": "file-word-2",
  "one": "file-word-2",
  "otf": "image",
  "pdf": "file-pdf",
  "pfb": "image",
  "php": "file-code",
  "pl": "file-code",
  "png": "image",
  "pot": "file-ppt-2",
  "potm": "file-ppt-2",
  "potx": "file-ppt-2",
  "ppa": "file-ppt-2",
  "ppam": "file-ppt-2",
  "pps": "file-ppt-2",
  "ppsm": "file-ppt-2",
  "ppsx": "file-ppt-2",
  "ppt": "file-ppt-2",
  "pptm": "file-ppt-2",
  "pptx": "file-ppt-2",
  "ps": "image",
  "psd": "image",
  "py": "file-code",
  "rar": "file-zip",
  "rss": "file",
  "sh-lib": "file-code",
  "sh": "file-code",
  "swf": "file",
  "tar": "file-zip",
  "tar.bz2": "file-zip",
  "tar.gz": "file-zip",
  "tbz2": "file-zip",
  "tex": "file-text",
  "tgz": "file-zip",
  "ttf": "image",
  "vsd": "file-word-2",
  "woff": "image",
  "wpd": "file-word-2",
  "xcf": "image",
  "xla": "file-excel-2",
  "xlam": "file-excel-2",
  "xls": "file-excel-2",
  "xlsb": "file-excel-2",
  "xlsm": "file-excel-2",
  "xlsx": "file-excel-2",
  "xlt": "file-excel-2",
  "xltm": "file-excel-2",
  "xltx": "file-excel-2",
  "xml": "file-code",
  "yaml": "file-code",
  "yml": "file-code",
  "zip": "file-zip"
}
