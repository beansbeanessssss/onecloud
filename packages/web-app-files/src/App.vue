<template>
  <main id="files" class="oc-flex oc-height-1-1">
    <div v-if="dragareaEnabled" class="dragarea" />
    <router-view tabindex="0" class="oc-width-expand" />
  </main>
</template>
<script lang="ts">
import { defineComponent, watch } from '@vue/composition-api'
import { Route } from 'vue-router'
import { useRoute, useStore } from 'web-pkg/src/composables'
import { bus } from 'web-pkg/src/instance'

export default defineComponent({
  setup() {
    const store = useStore<any>()
    watch(useRoute(), (to: Route, from?: Route) => {
      store.dispatch('Files/resetFileSelection')
    })
  },
  data: () => ({
    dragareaEnabled: false
  }),
  created() {
    const dragOver = bus.subscribe('drag-over', this.onDragOver)
    const dragOut = bus.subscribe('drag-out', this.hideDropzone)
    const drop = bus.subscribe('drop', this.hideDropzone)

    this.$on('beforeDestroy', () => {
      bus.unsubscribe('drag-over', dragOver)
      bus.unsubscribe('drag-out', dragOut)
      bus.unsubscribe('drop', drop)
    })
  },

  methods: {
    hideDropzone() {
      this.dragareaEnabled = false
    },
    onDragOver(event) {
      this.dragareaEnabled = (event.dataTransfer.types || []).some((e) => e === 'Files')
    }
  }
})
</script>

<style lang="scss" scoped>
main {
  max-height: 100%;
}

.dragarea {
  background-color: rgba(60, 130, 225, 0.21);
  pointer-events: none;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  position: absolute;
  z-index: 9;
  border-radius: 14px;
  border: 2px dashed var(--oc-color-swatch-primary-muted);
}

#files {
  position: relative;
}
</style>
