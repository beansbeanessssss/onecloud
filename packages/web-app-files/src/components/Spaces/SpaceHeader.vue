<template>
  <div class="space-header" :class="{ 'oc-flex': imageContent && !imageExpanded }">
    <div v-if="imageContent" :class="{ 'oc-width-1-4 oc-mr-l': !imageExpanded }">
      <img
        :class="{ expanded: imageExpanded }"
        class="space-header-image oc-cursor-pointer"
        alt=""
        :src="imageContent"
        @click="toggleImageExpanded"
      />
    </div>
    <div :class="{ 'oc-width-3-4': imageContent && !imageExpanded }">
      <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
        <div class="oc-flex oc-flex-middle">
          <h1 class="space-header-name oc-text-truncate">{{ space.name }}</h1>
          <oc-button
            :id="`space-context-btn`"
            v-oc-tooltip="$gettext('Show context menu')"
            :aria-label="$gettext('Show context menu')"
            appearance="raw"
            class="oc-ml-s"
          >
            <oc-icon name="more-2" />
          </oc-button>
          <oc-drop
            :drop-id="`space-context-drop`"
            :toggle="`#space-context-btn`"
            mode="click"
            close-on-click
            :options="{ delayHide: 0 }"
            padding-size="small"
            position="right-start"
          >
            <space-context-actions :items="[space]" :space="space" />
          </oc-drop>
        </div>
        <oc-button
          v-if="memberCount"
          :aria-label="$gettext('Open context menu and show members')"
          appearance="raw"
          @click="openSideBarSharePanel"
        >
          <oc-icon name="group" fill-type="line" size="small" />
          <span class="space-header-people-count oc-text-small" v-text="memberCountString"></span>
        </oc-button>
      </div>
      <p v-if="space.description" class="oc-mt-rm">{{ space.description }}</p>
      <div>
        <!-- eslint-disable vue/no-v-html -->
        <div ref="markdownContainerRef" class="markdown-container" v-html="markdownContent"></div>
        <!-- eslint-enable -->
        <div v-if="showMarkdownCollapse" class="markdown-collapse oc-text-center oc-mt-s">
          <oc-button appearance="raw" @click="toggleMarkdownCollapsed">
            <oc-icon :name="toggleMarkdownCollapsedIcon" />
            <span>{{ toggleMarkdownCollapsedText }}</span>
          </oc-button>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import {
  computed,
  defineComponent,
  onBeforeUnmount,
  onMounted,
  PropType,
  ref,
  unref,
  watch
} from '@vue/composition-api'
import { SpaceResource } from 'web-client/src/helpers'
import { loadPreview } from 'web-pkg/src/helpers'
import {
  useAccessToken,
  useClientService,
  useStore,
  useTranslations
} from 'web-pkg/src/composables'
import { ImageDimension } from '../../constants'
import { configurationManager } from 'web-pkg/src/configuration'
import { VisibilityObserver } from 'web-pkg/src/observer'
import { marked } from 'marked'
import sanitizeHtml from 'sanitize-html'
import SpaceContextActions from './SpaceContextActions.vue'
import { bus } from 'web-pkg/src/instance'
import { SideBarEventTopics } from '../../composables/sideBar'

const visibilityObserver = new VisibilityObserver()
const markdownContainerCollapsedClass = 'collapsed'

export default defineComponent({
  name: 'SpaceHeader',
  components: {
    SpaceContextActions
  },
  props: {
    space: {
      type: Object as PropType<SpaceResource>,
      required: true
    }
  },
  setup(props) {
    const { $gettext, $ngettext, $gettextInterpolate } = useTranslations()
    const { getFileContents, getFileInfo } = useClientService().webdav
    const store = useStore()
    const userId = computed(() => store.getters.user?.id)
    const accessToken = useAccessToken({ store })

    const markdownContainerRef = ref(null)
    const markdownContent = ref('')
    const markdownCollapsed = ref(true)
    const showMarkdownCollapse = ref(false)
    const toggleMarkdownCollapsedIcon = computed(() => {
      return unref(markdownCollapsed) ? 'add' : 'subtract'
    })
    const toggleMarkdownCollapsedText = computed(() => {
      return unref(markdownCollapsed) ? $gettext('Show more') : $gettext('Show less')
    })
    const toggleMarkdownCollapsed = () => {
      markdownCollapsed.value = !unref(markdownCollapsed)
      unref(markdownContainerRef).classList.toggle(markdownContainerCollapsedClass)
    }
    const onMarkdownResize = () => {
      if (!unref(markdownContainerRef)) {
        return
      }

      unref(markdownContainerRef).classList.remove(markdownContainerCollapsedClass)
      const markdownContainerHeight = unref(markdownContainerRef).offsetHeight
      if (markdownContainerHeight < 150) {
        showMarkdownCollapse.value = false
        return
      }
      showMarkdownCollapse.value = true

      if (unref(markdownCollapsed)) {
        unref(markdownContainerRef).classList.add(markdownContainerCollapsedClass)
      }
    }
    const markdownResizeObserver = new ResizeObserver(onMarkdownResize)
    const observeMarkdownContainerResize = () => {
      if (!markdownResizeObserver || !unref(markdownContainerRef)) {
        return
      }
      markdownResizeObserver.unobserve(unref(markdownContainerRef))
      markdownResizeObserver.observe(unref(markdownContainerRef))
    }
    const unobserveMarkdownContainerResize = () => {
      if (!markdownResizeObserver || !unref(markdownContainerRef)) {
        return
      }
      markdownResizeObserver.unobserve(unref(markdownContainerRef))
    }
    onMounted(observeMarkdownContainerResize)
    onBeforeUnmount(() => {
      visibilityObserver.disconnect()
      unobserveMarkdownContainerResize()
    })
    watch(
      computed(() => props.space.spaceReadmeData),
      async (data: any) => {
        if (!data) {
          return
        }
        const decodedUri = decodeURI(data.webDavUrl)
        const webDavPathComponents = decodedUri.split('/')
        const idComponent = webDavPathComponents.find((c) => c.startsWith(props.space.id as string))
        if (!idComponent) {
          return
        }
        const path = webDavPathComponents
          .slice(webDavPathComponents.indexOf(idComponent) + 1)
          .join('/')

        const fileContentsResponse = await getFileContents(props.space, {
          path: decodeURIComponent(path)
        })

        unobserveMarkdownContainerResize()
        const parsedMarkdown = marked.parse(fileContentsResponse.body)
        // Sanitize markdown content to prevent XSS vulnerabilities
        markdownContent.value = sanitizeHtml(parsedMarkdown)

        if (unref(markdownContent)) {
          observeMarkdownContainerResize()
        }
      },
      { deep: true, immediate: true }
    )

    const imageContent = ref(null)
    const imageExpanded = ref(false)
    const toggleImageExpanded = () => {
      imageExpanded.value = !unref(imageExpanded)
    }
    watch(
      computed(() => props.space.spaceImageData),
      async (data) => {
        if (!data) {
          return
        }
        const decodedUri = decodeURI(props.space.spaceImageData.webDavUrl)
        const webDavPathComponents = decodedUri.split('/')
        const idComponent = webDavPathComponents.find((c) => c.startsWith(props.space.id as string))
        if (!idComponent) {
          return
        }
        const path = webDavPathComponents
          .slice(webDavPathComponents.indexOf(idComponent) + 1)
          .join('/')

        const resource = await getFileInfo(props.space, {
          path: decodeURIComponent(path)
        })
        imageContent.value = await loadPreview({
          resource,
          isPublic: false,
          dimensions: ImageDimension.Preview,
          server: configurationManager.serverUrl,
          userId: unref(userId),
          token: unref(accessToken)
        })
      },
      { deep: true, immediate: true }
    )

    const memberCount = computed(() => {
      return store.getters['runtime/spaces/spaceMembers'].length
    })
    const memberCountString = computed(() => {
      const translated = $ngettext('%{count} member', '%{count} members', unref(memberCount))

      return $gettextInterpolate(translated, {
        count: unref(memberCount)
      })
    })

    const openSideBarSharePanel = () => {
      store.commit('Files/SET_SELECTED_IDS', [])
      bus.publish(SideBarEventTopics.openWithPanel, 'space-share-item')
    }

    return {
      markdownContainerRef,
      markdownContent,
      markdownCollapsed,
      showMarkdownCollapse,
      toggleMarkdownCollapsedIcon,
      toggleMarkdownCollapsedText,
      toggleMarkdownCollapsed,
      imageContent,
      imageExpanded,
      toggleImageExpanded,
      memberCount,
      memberCountString,
      openSideBarSharePanel
    }
  }
})
</script>

<style lang="scss">
.space-header {
  &-image {
    border-radius: 10px;
    width: 100%;
    aspect-ratio: 16 / 9;
    object-fit: cover;
  }

  &-image.expanded {
    max-height: 100%;
    max-width: 100%;
  }

  &-name {
    font-size: 1.5rem;
  }

  &-people-count {
    white-space: nowrap;
  }

  .markdown-container.collapsed {
    max-height: 150px;
    overflow: hidden;
    -webkit-mask-image: linear-gradient(180deg, #000 90%, transparent);
  }
}
</style>
