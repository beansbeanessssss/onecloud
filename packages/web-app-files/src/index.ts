import App from './App.vue'
import Favorites from './views/Favorites.vue'
import FilesDrop from './views/FilesDrop.vue'
import SharedWithMe from './views/shares/SharedWithMe.vue'
import SharedWithOthers from './views/shares/SharedWithOthers.vue'
import SharedViaLink from './views/shares/SharedViaLink.vue'
import SpaceDriveResolver from './views/spaces/DriveResolver.vue'
import SpaceProjects from './views/spaces/Projects.vue'
import TrashOverview from './views/trash/Overview.vue'
import translations from '../l10n/translations.json'
import { defineWebApplication, quickActions } from '@ownclouders/web-pkg'
import store from './store'
import { extensions } from './extensions'
import fileSideBars from './fileSideBars'
import { buildRoutes } from '@ownclouders/web-pkg'
import { AppNavigationItem } from '@ownclouders/web-pkg'

// dirty: importing view from other extension within project
import SearchResults from '../../web-app-search/src/views/List.vue'
import {
  SpaceResource,
  isPersonalSpaceResource,
  isShareSpaceResource
} from '@ownclouders/web-client/src/helpers'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Files'),
  id: 'files',
  icon: 'resource-type-folder',
  color: 'var(--oc-color-swatch-primary-muted)',
  isFileEditor: false,
  extensions: [],
  fileSideBars
}
export const navItems = (context): AppNavigationItem[] => {
  return [
    {
      name(capabilities) {
        return capabilities.spaces?.enabled ? $gettext('Personal') : $gettext('All files')
      },
      icon: appInfo.icon,
      route: {
        path: `/${appInfo.id}/spaces/personal`
      },
      isActive: () => {
        return (
          !context.$store.getters['runtime/spaces/currentSpace'] ||
          context.$store.getters['runtime/spaces/currentSpace']?.isOwner(
            context.$store.getters.user
          )
        )
      },
      enabled(capabilities) {
        if (!capabilities.spaces?.enabled) {
          return true
        }
        return !!context?.$store?.getters['runtime/spaces/spaces'].find(
          (drive) => isPersonalSpaceResource(drive) && drive.isOwner(context.$store.getters.user)
        )
      }
    },
    {
      name: $gettext('Favorites'),
      icon: 'star',
      route: {
        path: `/${appInfo.id}/favorites`
      },
      enabled(capabilities) {
        return capabilities.files?.favorites
      }
    },
    {
      name: $gettext('Shares'),
      icon: 'share-forward',
      route: {
        path: `/${appInfo.id}/shares`
      },
      isActive: () => {
        const space = context.$store.getters['runtime/spaces/currentSpace'] as SpaceResource
        // last check is when fullShareOwnerPaths is enabled
        return !space || isShareSpaceResource(space) || !space?.isOwner(context.$store.getters.user)
      },
      activeFor: [
        { path: `/${appInfo.id}/spaces/share` },
        { path: `/${appInfo.id}/spaces/personal` }
      ],
      enabled(capabilities) {
        return capabilities.files_sharing?.api_enabled !== false
      }
    },
    {
      name: $gettext('Spaces'),
      icon: 'layout-grid',
      route: {
        path: `/${appInfo.id}/spaces/projects`
      },
      activeFor: [{ path: `/${appInfo.id}/spaces/project` }],
      enabled(capabilities) {
        return capabilities.spaces?.projects
      }
    },
    {
      name: $gettext('Deleted files'),
      icon: 'delete-bin-5',
      route: {
        path: `/${appInfo.id}/trash/overview`
      },
      activeFor: [{ path: `/${appInfo.id}/trash` }],
      enabled(capabilities) {
        return capabilities.dav?.trashbin === '1.0' && capabilities.files?.undelete
      }
    }
  ]
}

export default defineWebApplication({
  setup(args) {
    return {
      appInfo,
      store,
      routes: buildRoutes({
        App,
        Favorites,
        FilesDrop,
        SearchResults,
        Shares: {
          SharedViaLink,
          SharedWithMe,
          SharedWithOthers
        },
        Spaces: {
          DriveResolver: SpaceDriveResolver,
          Projects: SpaceProjects
        },
        Trash: {
          Overview: TrashOverview
        }
      }),
      navItems,
      quickActions,
      translations,
      extensions: extensions(args)
    }
  }
})
