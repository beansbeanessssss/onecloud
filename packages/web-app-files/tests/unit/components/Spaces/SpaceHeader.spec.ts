import SpaceHeader from 'web-app-files/src/components/Spaces/SpaceHeader.vue'
import { buildSpace } from 'web-client/src/helpers'
import {
  createStore,
  defaultPlugins,
  mount,
  defaultStoreMockOptions,
  defaultComponentMocks
} from 'web-test-helpers'

window.ResizeObserver =
  window.ResizeObserver ||
  jest.fn().mockImplementation(() => ({
    disconnect: jest.fn(),
    observe: jest.fn(),
    unobserve: jest.fn()
  }))

afterEach(() => jest.clearAllMocks())

describe('SpaceHeader', () => {
  it('should add the "squashed"-class when the sidebar is opened', () => {
    const wrapper = getWrapper({ space: buildSpace({ id: 1 }), sideBarOpen: true })
    expect(wrapper.find('.space-header-squashed').exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('space image', () => {
    it('should show the default image if no other image is set', () => {
      const wrapper = getWrapper({ space: buildSpace({ id: 1 }) })
      expect(wrapper.find('.space-header-image-default').exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('should show the set image', () => {
      const spaceMock = { spaceImageData: { webDavUrl: '/' } }
      const wrapper = getWrapper({ space: { ...buildSpace({ id: 1 }), ...spaceMock } })
      expect(wrapper.find('.space-header-image-default').exists()).toBeFalsy()
      expect(wrapper.find('.space-header-image img').exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})

function getWrapper({ space = {}, sideBarOpen = false }) {
  const mocks = {
    ...defaultComponentMocks(),
    $permissionManager: {
      canEditSpaceQuota: () => true
    }
  }
  const store = createStore(defaultStoreMockOptions)
  return mount(SpaceHeader, {
    props: {
      space,
      sideBarOpen
    },
    global: {
      mocks,
      plugins: [...defaultPlugins(), store],
      stubs: {
        'quota-modal': true
      }
    }
  })
}
