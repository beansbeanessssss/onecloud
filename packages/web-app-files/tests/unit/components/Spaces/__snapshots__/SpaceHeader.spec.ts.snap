// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceHeader should add the "squashed"-class when the sidebar is opened 1`] = `
<div class="space-header oc-p-m oc-flex space-header-squashed">
  <div class="space-header-image">
    <div class="space-header-image-default oc-flex oc-flex-middle oc-flex-center">
      <span class="oc-icon oc-icon-xxl oc-icon-passive oc-px-m oc-py-m">
        <!---->
      </span>
    </div>
  </div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h1 class="space-header-name"></h1>
        <button aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-ml-s" id="space-context-btn" type="button">
          <!-- @slot Content of the button -->
          <span class="oc-icon oc-icon-m oc-icon-passive">
            <!---->
          </span>
        </button>
        <div class="oc-drop oc-box-shadow-medium oc-rounded" id="space-context-drop" options="[object Object]">
          <div class="oc-card oc-card-body oc-background-secondary oc-p-s">
            <div>
              <div id="oc-files-context-menu">
                <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pb-s oc-files-context-actions-border" id="oc-files-context-actions-members">
                  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover">
                    <button class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-details-trigger action-menu-item" type="button">
                      <!-- @slot Content of the button -->
                      <span class="oc-icon oc-icon-m oc-icon-passive">
                        <!---->
                      </span>
                      <span class="oc-files-context-action-label">Members</span>
                      <!--v-if-->
                      <!--v-if-->
                    </button>
                  </li>
                </ul>
                <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pt-s" id="oc-files-context-actions-trashBin">
                  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover">
                    <button class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-delete-trigger action-menu-item" type="button">
                      <!-- @slot Content of the button -->
                      <span class="oc-icon oc-icon-m oc-icon-passive">
                        <!---->
                      </span>
                      <span class="oc-files-context-action-label">Deleted files</span>
                      <!--v-if-->
                      <!--v-if-->
                    </button>
                  </li>
                </ul>
              </div>
              <!--v-if-->
              <!--v-if-->
              <input accept="image/*" id="space-image-upload-input" multiple="" name="file" tabindex="-1" type="file">
            </div>
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <div>
      <!-- eslint-disable vue/no-v-html -->
      <div class="markdown-container"></div> <!-- eslint-enable -->
      <!--v-if-->
    </div>
  </div>
</div>
`;

exports[`SpaceHeader space image should show the default image if no other image is set 1`] = `
<div class="space-header oc-p-m oc-flex">
  <div class="space-header-image">
    <div class="space-header-image-default oc-flex oc-flex-middle oc-flex-center">
      <span class="oc-icon oc-icon-xxl oc-icon-passive oc-px-m oc-py-m">
        <!---->
      </span>
    </div>
  </div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h1 class="space-header-name"></h1>
        <button aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-ml-s" id="space-context-btn" type="button">
          <!-- @slot Content of the button -->
          <span class="oc-icon oc-icon-m oc-icon-passive">
            <!---->
          </span>
        </button>
        <div class="oc-drop oc-box-shadow-medium oc-rounded" id="space-context-drop" options="[object Object]">
          <div class="oc-card oc-card-body oc-background-secondary oc-p-s">
            <div>
              <div id="oc-files-context-menu">
                <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pb-s oc-files-context-actions-border" id="oc-files-context-actions-members">
                  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover">
                    <button class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-details-trigger action-menu-item" type="button">
                      <!-- @slot Content of the button -->
                      <span class="oc-icon oc-icon-m oc-icon-passive">
                        <!---->
                      </span>
                      <span class="oc-files-context-action-label">Members</span>
                      <!--v-if-->
                      <!--v-if-->
                    </button>
                  </li>
                </ul>
                <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pt-s" id="oc-files-context-actions-trashBin">
                  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover">
                    <button class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-delete-trigger action-menu-item" type="button">
                      <!-- @slot Content of the button -->
                      <span class="oc-icon oc-icon-m oc-icon-passive">
                        <!---->
                      </span>
                      <span class="oc-files-context-action-label">Deleted files</span>
                      <!--v-if-->
                      <!--v-if-->
                    </button>
                  </li>
                </ul>
              </div>
              <!--v-if-->
              <!--v-if-->
              <input accept="image/*" id="space-image-upload-input" multiple="" name="file" tabindex="-1" type="file">
            </div>
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <div>
      <!-- eslint-disable vue/no-v-html -->
      <div class="markdown-container"></div> <!-- eslint-enable -->
      <!--v-if-->
    </div>
  </div>
</div>
`;

exports[`SpaceHeader space image should show the set image 1`] = `
<div class="space-header oc-p-m oc-flex">
  <div class="space-header-image">
    <img alt="" class="oc-cursor-pointer">
  </div>
  <div class="space-header-infos">
    <div class="oc-flex oc-mb-s oc-flex-middle oc-flex-between">
      <div class="oc-flex oc-flex-middle space-header-infos-heading">
        <h1 class="space-header-name"></h1>
        <button aria-label="Show context menu" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-ml-s" id="space-context-btn" type="button">
          <!-- @slot Content of the button -->
          <span class="oc-icon oc-icon-m oc-icon-passive">
            <!---->
          </span>
        </button>
        <div class="oc-drop oc-box-shadow-medium oc-rounded" id="space-context-drop" options="[object Object]">
          <div class="oc-card oc-card-body oc-background-secondary oc-p-s">
            <div>
              <div id="oc-files-context-menu">
                <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pb-s oc-files-context-actions-border" id="oc-files-context-actions-members">
                  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover">
                    <button class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-show-details-trigger action-menu-item" type="button">
                      <!-- @slot Content of the button -->
                      <span class="oc-icon oc-icon-m oc-icon-passive">
                        <!---->
                      </span>
                      <span class="oc-files-context-action-label">Members</span>
                      <!--v-if-->
                      <!--v-if-->
                    </button>
                  </li>
                </ul>
                <ul class="oc-list oc-my-rm oc-mx-rm oc-files-context-actions oc-pt-s" id="oc-files-context-actions-trashBin">
                  <li class="context-menu oc-files-context-action oc-px-s oc-rounded oc-menu-item-hover">
                    <button class="oc-button oc-rounded oc-button-s oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-files-actions-delete-trigger action-menu-item" type="button">
                      <!-- @slot Content of the button -->
                      <span class="oc-icon oc-icon-m oc-icon-passive">
                        <!---->
                      </span>
                      <span class="oc-files-context-action-label">Deleted files</span>
                      <!--v-if-->
                      <!--v-if-->
                    </button>
                  </li>
                </ul>
              </div>
              <!--v-if-->
              <!--v-if-->
              <input accept="image/*" id="space-image-upload-input" multiple="" name="file" tabindex="-1" type="file">
            </div>
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--v-if-->
    <div>
      <!-- eslint-disable vue/no-v-html -->
      <div class="markdown-container"></div> <!-- eslint-enable -->
      <!--v-if-->
    </div>
  </div>
</div>
`;
